### PR TITLE
vscode: add a script to generate debugger config

### PR DESCRIPTION
**Pull request by lizan**
Commit Message:
Additional Description:
Risk Level: Low
Testing: local
Docs Changes: Added
Release Notes: N/A

### CODE REVIEW

**Comment by lizan** [comment]
cc @davinci26

File: tools/vscode/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+## Generate debug config
+
+`tools/vscode/generate_debug_config.py` is a script to generate VSCode debug config in `.vscode/launch.json`.
+
+For example:
+```
+tools/vscode/generate_debug_config.py //source/exe:envoy-static --args "-c envoy.yaml"
+```
+
+Generates an entry for gdb in `launch.json`. It can be also used to generate config for tests.
```
Comment (snowp) [inline tools/vscode/README.md:30]: Maybe include a note about what the name of the generated task will be?

Comment (lizan) [reply tools/vscode/README.md]: Added.

**Comment by lizan** [comment]
> Also, out of scope, but I wonder if it is possible and what it would take to pre-generate a populated picker list based on this with all targets

That probably need to be a VS Code extension...

**Comment by wrowe** [comment]
Some of the folks involved with @envoyproxy/windows-dev or @rmiller14 may be knowledgeable in this subject.

File: tools/vscode/generate_debug_config.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  return subprocess.check_output(["bazel", "info", name] + BAZEL_OPTIONS +
+                                 bazel_extra_options).decode().strip()
+
+
+def getWorkspace():
+  return bazelInfo("workspace")
+
+
+def getExecutionRoot(workspace):
+  # If compilation database exists, use its execution root
```
Comment (snowp) [inline tools/vscode/generate_debug_config.py:24]: Explaining why the comp db execution root is preferred might be nice

**Review by snowp — CHANGES_REQUESTED** [review]
Just one nit, otherwise LGTM

**Review by snowp — APPROVED** [review]
Thanks!

**Pull request merged**